src/EventListener/AssetsInstallCommandListener.php

```diff
@@ -0,0 +1,260 @@
+    {
+        /**
+         * @var KernelInterface
+         */
+        $kernel = $application->getKernel();
+
+        $targetArg = rtrim($input->getArgument('target') ?? '', '/');
+
+        if (!$targetArg) {
+            $targetArg = $this->getPublicDirectory($kernel->getContainer());
+        }
+
+        if (!is_dir($targetArg)) {
+            $targetArg = $kernel->getProjectDir().'/'.$targetArg;
+
+            if (!is_dir($targetArg)) {
+                throw new InvalidArgumentException(sprintf('The target directory "%s" does not exist.', $input->getArgument('target')));
+            }
+        }
+
+        $bundlesDir = $targetArg.'/bundles/';
+
+        $io = new SymfonyStyle($input, $output);
+        $io->newLine();
+
+        if ($input->getOption('relative')) {
+            $expectedMethod = self::METHOD_RELATIVE_SYMLINK;
+            $io->text('Trying to install deprecated SonataCoreBundle assets from SonataAdminBundle as <info>relative symbolic links</info>.');
+        } elseif ($input->getOption('symlink')) {
+            $expectedMethod = self::METHOD_ABSOLUTE_SYMLINK;
+            $io->text('Trying to install deprecated SonataCoreBundle assets from SonataAdminBundle as <info>absolute symbolic links</info>.');
+        } else {
+            $expectedMethod = self::METHOD_COPY;
+            $io->text('Installing deprecated SonataCoreBundle assets from SonataAdminBundle as <info>hard copies</info>.');
+        }
+
+        $io->newLine();
+
+        $copyUsed = false;
+        $exitCode = 0;
+        $validAssetDirs = [];
+
+        $bundle = $kernel->getBundle('SonataAdminBundle');
+        $originDir = $bundle->getPath().'/Resources/private/SonataCoreBundleAssets';
+
+        $assetDir = preg_replace('/bundle$/', '', 'sonatacore');
+        $targetDir = $bundlesDir.$assetDir;
+        $validAssetDirs[] = $assetDir;
+
+        if (OutputInterface::VERBOSITY_VERBOSE <= $output->getVerbosity()) {
+            $message = sprintf("%s\n-> %s", $bundle->getName(), $targetDir);
+        } else {
+            $message = $bundle->getName();
+        }
+
+        try {
+            $this->filesystem->remove($targetDir);
+
+            if (self::METHOD_RELATIVE_SYMLINK === $expectedMethod) {
+                $method = $this->relativeSymlinkWithFallback($originDir, $targetDir);
+            } elseif (self::METHOD_ABSOLUTE_SYMLINK === $expectedMethod) {
+                $method = $this->absoluteSymlinkWithFallback($originDir, $targetDir);
+            } else {
+                $method = $this->hardCopy($originDir, $targetDir);
+            }
+
+            if (self::METHOD_COPY === $method) {
+                $copyUsed = true;
+            }
+
+            if ($method === $expectedMethod) {
+                $ioMethod = 'success';
+            } else {
+                $ioMethod = 'warning';
+            }
+        } catch (\Exception $e) {
```

Comment:
Shouldn't we catch `\Throwable`?

Reply:
Here not. It is copy from symfony assets:install command.